src/main/java/com/cleanroommc/groovyscript/compat/mods/prodigytech/Solderer.java

```diff
@@ -0,0 +1,160 @@
+package com.cleanroommc.groovyscript.compat.mods.prodigytech;
+
+import com.cleanroommc.groovyscript.api.GroovyLog;
+import com.cleanroommc.groovyscript.api.IIngredient;
+import com.cleanroommc.groovyscript.api.documentation.annotations.*;
+import com.cleanroommc.groovyscript.compat.mods.ModSupport;
+import com.cleanroommc.groovyscript.helper.SimpleObjectStream;
+import com.cleanroommc.groovyscript.helper.recipe.AbstractRecipeBuilder;
+import com.cleanroommc.groovyscript.registry.VirtualizedRegistry;
+import lykrast.prodigytech.common.recipe.SoldererManager;
+import lykrast.prodigytech.common.util.Config;
+import net.minecraft.item.ItemStack;
+import org.jetbrains.annotations.Nullable;
+
+@RegistryDescription
+public class Solderer extends VirtualizedRegistry<SoldererManager.SoldererRecipe> {
+
+    @RecipeBuilderDescription(example = {
+        @Example(".pattern(item('minecraft:clay')).input(item('minecraft:gold_ingot')).output(item('minecraft:diamond')).gold(5).time(100)"),
+        @Example(".pattern(item('minecraft:coal_block')).output(item('minecraft:nether_star')).gold(75)"),
+    })
+    public Solderer.RecipeBuilder recipeBuilder() {
+        return new Solderer.RecipeBuilder();
+    }
+
+    @Override
+    public void onReload() {
+        removeScripted().forEach(x -> SoldererManager.removeRecipe(x.getPattern(), x.getAdditive(), 9999));
+        restoreFromBackup().forEach(SoldererManager::addRecipe);
+    }
+
+    public void add(SoldererManager.SoldererRecipe recipe) {
+        if (recipe == null) return;
+        addScripted(recipe);
+        SoldererManager.addRecipe(recipe);
+    }
+
+    public boolean remove(SoldererManager.SoldererRecipe recipe) {
+        if (recipe == null) return false;
+        addBackup(recipe);
+        return SoldererManager.removeRecipe(recipe.getPattern(), recipe.getAdditive(), 9999) != null;
+    }
+
+    @MethodDescription(example = @Example("item('prodigytech:pattern_circuit_refined')"))
+    public boolean removeByPattern(IIngredient pattern) {
+        SoldererManager.SoldererRecipe recipe = SoldererManager.RECIPES.stream()
+                .filter(r -> pattern.test(r.getPattern()))
+                .findFirst().orElse(null);
+        return remove(recipe);

```

Comment:
i dont see the need to do this, you should be able to run this directly
```suggestion
        return SoldererManager.RECIPES.removeIf(r -> {
            if (pattern.test(r.getPattern())) {
                addBackup(r);
                return true;
            }
            return false;
        });
```